lib/sections/driver.js

```diff
@@ -1,60 +1,89 @@
 var utils = require(__dirname + '/utils.js');
 
-var driverJob = function(){
+var busJob = function(){
   return {
     plugins: [],
     pluginOptions: [],
     perform: function(args, callback){
       var self = this;
 
-      var started = 0;
-      var matches = [];
-      self.subscriptions(function(err, subscriptions, count){
-        if(count == 0){
-          if(started === 0 && typeof callback === 'function'){ callback(null, matches); }
-        }else{    
-          for(var app in subscriptions){
-            for(var i in subscriptions[app]){
-              started++;
-              var subscription = subscriptions[app][i];
-              var matched = subscriptionMatch(args, subscription);
-              if(matched === true){
-                matches.push(subscription['queue_name']);
-                var payload = driverMetadata(args, subscription.queue_name, app, i, subscription.class, args.bus_event_type);
-                self.queueObject.enqueue(subscription.queue_name, subscription.class, payload, function(err, toRun){
-                  started--;
-                  if(started === 0 && typeof callback === 'function'){ callback(null, matches); }
-                });
-              }else{
-                process.nextTick(function(){
-                  started--;
-                  if(started === 0 && typeof callback === 'function'){ callback(null, matches); }
-                });
-              }
-            }
+      if(args.bus_class_proxy === '::QueueBus::Driver'){   
+        driverPerform(self, args, callback); 
+      }else if(args.bus_class_proxy === '::QueueBus::Rider'){    
+        callback(new Error('TODO: ::QueueBus::Rider')); // TODO!
+      }else if(args.bus_class_proxy === '::QueueBus::Publisher'){
+        publisherPerform(self, args, callback); 
+      }else if(args.bus_class_proxy === '::QueueBus::Heartbeat'){
+        callback(new Error('TODO: ::QueueBus::Heartbeat')); // TODO!
+      }else{
+        callback(new Error('Missing bus_class_proxy or unknown method (' + args.bus_class_proxy + ')'));
+      }
```

Comment:
It's set up in resque scheduler to run every minute. It also keeps track (in redis) of when it last published by incrementing some number or something and it works on the "now" time. That is, it might publish two or more events if it gets behind.